src/libraries/System.Text.Json/tests/Serialization/ConstructorTests/ConstructorTests.Exceptions.cs

```diff
@@ -129,18 +140,81 @@ public class Class_ExtData_CtorParam
         }
 
         [Fact]
-        public void AnonymousObject_InvalidOperationException()
+        public void AnonymousObject()
         {
             var obj = new { Prop = 5 };
+            Type objType = obj.GetType();
 
-            InvalidOperationException ex = Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize("{}", obj.GetType()));
+            // 'Prop' property binds with a ctor arg called 'Prop'.
 
-            // We expect property 'Prop' to bind with a ctor arg called 'prop', but the ctor arg is called 'Prop'.
-            string exStr = ex.ToString();
-            Assert.Contains("AnonymousType", exStr);
-            Assert.Contains("(Int32)", exStr);
-            Assert.Contains("[System.Int32]", exStr);
+            object newObj = JsonSerializer.Deserialize("{}", objType);
+            Assert.Equal(0, objType.GetProperty("Prop").GetValue(newObj));
+
+            newObj = JsonSerializer.Deserialize(@"{""Prop"":5}", objType);
+            Assert.Equal(5, objType.GetProperty("Prop").GetValue(newObj));
+        }
+
+        /* Enable when C# 9.0 'records' feature works in CI
```

Comment:
Can we file an issue to track this?

cc @safern do you know when we'll enable this in CI?

Reply:
It appears the performance CI leg for some reason is not using the same compiler\SDK as the other tests.

Reply:
@safern @DrewScoggins the "IsExternalInit" failures are [occurring ](https://dev.azure.com/dnceng/public/_build/results?buildId=727585&view=results)with the tests using the "record" type:

Net472
```
Serialization\ConstructorTests\ConstructorTests.Exceptions.cs(155,37): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [F:\workspace\_work\1\s\src\libraries\System.Text.Json\tests\System.Text.Json.Tests.csproj]
##[error]Serialization\ConstructorTests\ConstructorTests.Exceptions.cs(155,37): error CS0518: (NETCORE_ENGINEERING_TELEMETRY=Build) Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported
Serialization\ConstructorTests\ConstructorTests.Exceptions.cs(168,60): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [F:\workspace\_work\1\s\src\libraries\System.Text.Json\tests\System.Text.Json.Tests.csproj]
##[error]Serialization\ConstructorTests\ConstructorTests.Exceptions.cs(168,60): error CS0518: (NETCORE_ENGINEERING_TELEMETRY=Build) Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported
Serialization\ConstructorTests\ConstructorTests.Exceptions.cs(168,74): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [F:\workspace\_work\1\s\src\libraries\System.Text.Json\tests\System.Text.Json.Tests.csproj]
##[error]Serialization\ConstructorTests\ConstructorTests.Exceptions.cs(168,74): error CS0518: (NETCORE_ENGINEERING_TELEMETRY=Build) Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported
```

Reply:
It seems like `IsExternalInit` (Which the record keyword is translated to use that) doesn't exist in `net472`, so we should probably make these tests just be included on `NetCoreAppCurrent`.

cc: @jaredpar @stephentoub are there any plans on supporting `record` in downlevel TFMs? Maybe by an autogenerated internal attribute by the compiler?

Reply:
@safern 

> we should probably make these tests just be included on NetCoreAppCurrent.

Do you mean `#if NETCOREAPP`?

Reply:
Yeah or splitting the test file into two test files and include the netcoreapp one whenever `TargetFramework == $(NetCoreAppCurrent)`